### PR TITLE
CXP-811 Add new product web API ACL

### PR DESCRIPTION
**Pull request by romaindak**
**Description (for Contributor and Core Developer)**

image

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

**Review by tseho — COMMENTED** [review]
You forgot the integration test on the migration.
And you are not applying the ACL at all.

File: src/Akeneo/Tool/Bundle/ApiBundle/Security/AccessDeniedHandler.php [hunk 1]
```diff
@@ -57,4 +57,16 @@ private function getInflector(): Inflector
     {
         return new Inflector(new NoopWordInflector(), new NoopWordInflector());
     }
+
+    private function getActionName(Request $request): string
+    {
+        if (Request::METHOD_GET === $request->getMethod()) {
+            return 'list';
+        }
+        if (Request::METHOD_DELETE === $request->getMethod()) {
```
Comment (willy-ahva) [inline src/Akeneo/Tool/Bundle/ApiBundle/Security/AccessDeniedHandler.php:66]: A valider avec @fitn , cette modif va s'appliquer sur tous les endpoints de l'api. :warning: BC Break

File: tests/back/Pim/Enrichment/EndToEnd/Product/Product/ExternalApi/AbstractProductTestCase.php [hunk 2]
```diff
@@ -140,4 +145,21 @@ protected function assertSameProducts(array $expectedProduct, $identifier)
 
         Assert::assertSame($expectedProduct, $standardizedProduct);
     }
+
+    protected function deletePermissionAcl(string $aclPrivilegeIdentityId): void
```
Comment (willy-ahva) [inline tests/back/Pim/Enrichment/EndToEnd/Product/Product/ExternalApi/AbstractProductTestCase.php:149]: removeAclFromRole sounds better to me, what do you think? Or forbidPrivilegeToRole?

File: tests/back/Pim/Enrichment/EndToEnd/Product/Product/ExternalApi/DeleteProductEndToEnd.php [hunk 3]
```diff
@@ -40,6 +40,28 @@ public function testDeleteAProduct()
...
+    "code": 403,
+    "message": "Access forbidden. You are not allowed to delete products."
+}
+JSON;
+
+        $response = $client->getResponse();
+        $this->assertSame(Response::HTTP_FORBIDDEN, $response->getStatusCode());
+        $this->assertJsonStringEqualsJsonString($expectedResponse, $response->getContent());
+
+        $this->assertCount(7, $this->get('pim_catalog.repository.product')->findAll());
```
Comment (willy-ahva) [inline tests/back/Pim/Enrichment/EndToEnd/Product/Product/ExternalApi/DeleteProductEndToEnd.php:62]: Shouldn't it be removed? It does not seem to be necessary. What do you want to test with this?

**Review by willy-ahva — APPROVED** [review]

File: src/Akeneo/Tool/Bundle/ApiBundle/Security/AccessDeniedHandler.php (hunk 1 above)
Comment (romaindak) [reply src/Akeneo/Tool/Bundle/ApiBundle/Security/AccessDeniedHandler.php:66]: This change only updates the error message.

File: tests/back/Pim/Enrichment/EndToEnd/Product/Product/ExternalApi/AbstractProductTestCase.php (hunk 2 above)
Comment (romaindak) [reply tests/back/Pim/Enrichment/EndToEnd/Product/Product/ExternalApi/AbstractProductTestCase.php:149]: removeAclFromRole, why not :)

File: tests/back/Pim/Enrichment/EndToEnd/Product/Product/ExternalApi/DeleteProductEndToEnd.php (hunk 3 above)
Comment (romaindak) [reply tests/back/Pim/Enrichment/EndToEnd/Product/Product/ExternalApi/DeleteProductEndToEnd.php:62]: I check if the number of products is the same before and after the delete call.

**Review by RiffVentura — APPROVED** [review]

**Review by fitn — APPROVED** [review]

File: src/Akeneo/Tool/Bundle/ApiBundle/Security/AccessDeniedHandler.php (hunk 1 above)
Comment (fitn) [reply src/Akeneo/Tool/Bundle/ApiBundle/Security/AccessDeniedHandler.php:66]: This is ok for API :+1:

**Pull request merged**